src/main/java/training/endava/skeleton/SteamsApp.java

```diff
@@ -0,0 +1,79 @@
+package training.endava.skeleton;
+
+import jdk.internal.dynalink.linker.ConversionComparator;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.Year;
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class SteamsApp {
+    public static void main(String[] args) {
+        Person p1 = new Person(UUID.randomUUID(), "Popescu", "01234556789", "Iasi");
+        Person p2 = new Person(UUID.randomUUID(), "Ionescu", "01234556789", "Iasi");
+        Person p3 = new Person(UUID.randomUUID(), "Escu", "01234556789", "Iasi");
+        Person p4 = new Person(UUID.randomUUID(), "Liliescu", "01234556789", "Iasi");
+        Person p5 = new Person(UUID.randomUUID(), "Pop", "01234556789", "Iasi");
+        List<Person> persons1 = new ArrayList<>(Arrays.asList(p1, p2, p3, p4, p5));
+
+        Person p6 = new Person(UUID.randomUUID(), "Lincescu", "01234556789", "Iasi");
+        Person p7 = new Person(UUID.randomUUID(), "Liliescu", "01234556789", "Iasi");
+        Person p8 = new Person(UUID.randomUUID(), "Oprescu", "01234556789", "Iasi");
+        List<Person> persons2 = new ArrayList<>(Arrays.asList(p6, p7, p8));
+
+        Company c1 = new Company("Endava", persons1, 1992, CompanyArea.IT);
+        Company c2 = new Company("Regina Maria", persons2, 1990, CompanyArea.HEALTH);
+        List<Company> companies = new ArrayList<>(Arrays.asList(c1, c2));
+
+        companies.stream()
+                .map(Company::getName)
+                .forEach(System.out::println);
+        //I know it would be more efficient to do the filtering first, but...
+        companies.stream()
+                .mapToInt(Company::getFoundationYear)
+                .filter(year -> year > 1990)
+                .forEach(System.out::println);
+
+        companies.stream()
+                .min(Comparator.comparingInt(Company::getFoundationYear))
+                .ifPresent(System.out::println);
+
+        companies.stream()
+                .mapToInt(company -> company.getEmployees().size())
+                .reduce((size1, size2) -> size1 + size2)
```

Comment:
You could have used the signature that accepts an identity value, so that if there are no employees in the list, the number of persons would have been 0 not an empty optional, but it's up to you.